clang-tools-extra/test/clang-tidy/checkers/readability/use-builtin-literals-cpp23.cpp

```diff
@@ -0,0 +1,11 @@
+// RUN: %check_clang_tidy -std=c++23-or-later %s readability-use-builtin-literals %t
+
+#include <stddef.h>
+
+void warn_and_fix() {
+
+(size_t)6zu;
+// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: use builtin literals instead of casts [readability-use-builtin-literals]
+// CHECK-FIXES: 6uz;
```

Comment:
What's the difference between `zu` and `uz`? Will this also match `(size_t)6`? And `std::size_t`?

Reply:
It's just that the fix-it favors a specific order and casing for each sequence. It will, and I added some test cases to show that.